geometry/calipers_box.hpp

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "geometry/point2d.hpp"
+
+#include <vector>
+
+namespace m2
+{
+// When a size of a convex hull over a set of |points| is less than 4,
+// stores convex hull explicitly, otherwise stores smallest rectangle
+// containing the hull.
```

Comment:
Please add a note that the rectangle does not necessarily has its sides parallel to the coordinate axes.

Reply:
This box may need a mention that the rectangle stores its points in the counterclockwise order.

Reply:
Done